onyo/cli/rmdir.py

```diff
@@ -0,0 +1,62 @@
+from __future__ import annotations
+
+from pathlib import Path
+from typing import TYPE_CHECKING
+
+from onyo.lib.onyo import OnyoRepo
+from onyo.lib.commands import onyo_rmdir
+from onyo.lib.inventory import Inventory
+from onyo.shared_arguments import (
+    shared_arg_message,
+    shared_arg_no_auto_message,
+)
+
+if TYPE_CHECKING:
+    import argparse
+
+args_rmdir = {
+    'directory': dict(
+        metavar='PATH',
+        nargs='+',
+        help=r"""
+            Directories to delete; or assets to convert into an Asset File.
+        """
+    ),
+    'message': shared_arg_message,
+    'no_auto_message': shared_arg_no_auto_message,
+}
+
+epilog_rmdir = r"""
+.. rubric:: Examples
+
+Remove a user from a group:
+
+.. code:: shell
+
+    $ onyo rmdir accounting/Bingo\ Bob/
+
+Convert an empty Asset Directory into an Asset File:
+
+.. code:: shell
+
+    $ onyo rmdir accounting/Bingo\ Bob/laptop_apple_macbook.oiw629/
+
+"""
+
+
+def rmdir(args: argparse.Namespace) -> None:
+    r"""
+    Delete empty **DIRECTORY**\ s or convert empty Asset Directories into Asset
+    Files.
+
+    If the **DIRECTORY** does not exist, the path is protected, or the asset is
```

Comment:
> the path is protected

Does this include paths that are `.onyoignore`d?

Reply:
Probably? The support for `.onyoignore` is spotty across the codebase. It's present, but I don't think there's been a comprehensive audit yet. IMO, that's beyond the scope of this PR.

Reply:
Yes. Nothing works on `.onyoignore`'d stuff.

Reply:
This is covered in the respective `Inventory` methods.
`Inventory.remove_directory()` in this case.